bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/serialization/BoschServiceDataDeserializer.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.boschshc.internal.serialization;
+
+import java.lang.reflect.Type;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.boschshc.internal.devices.bridge.dto.DeviceServiceData;
+import org.openhab.binding.boschshc.internal.devices.bridge.dto.Scenario;
+import org.openhab.binding.boschshc.internal.services.dto.BoschSHCServiceState;
+
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+
+/**
+ * Utility class for JSON deserialization of device data and triggered scenarios using Google Gson.
+ *
+ * @author Patrick Gell - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class BoschServiceDataDeserializer implements JsonDeserializer<BoschSHCServiceState> {
```

Comment:
Our handler infrastructure and HTTP client already provide methods for deserialization. It should work out-of-the-box if the field names match the JSON property names. So it should not be necessary to write the deserialization code yourself. Or is there a specific reason why the generic mechanisms do not work?

Reply:
I have added the deserializer to be able to receive either `DeviceServiceData` or `Scenario`objects via the same LongPolling endpoint. Before there was only `DeviceServiceData` objects received and therefor the ArrayList in the `LongPollingResult` class typed with `DeviceServiceData`.
I changed the type to the super type `BoschSHCServiceState` but for that I needed the extra deserializer.

Reply:
ok, got it :+1: I will investigate a bit more whether this can be solved without a custom deserializer, but if I find no better solution this is fine. Thanks for the explanation 🙂